lib/Sema/CSBindings.cpp

```diff
@@ -1631,9 +1640,23 @@ void BindingSet::dump(llvm::raw_ostream &out, unsigned indent) const {
     attributes.push_back("subtype_of_existential");
   auto literalKind = getLiteralKind();
   if (literalKind != LiteralBindingKind::None) {
-    auto literalAttrStr = ("[literal: " + getLiteralBindingKind(literalKind)
-                        + "]").str();
-    attributes.push_back(literalAttrStr);
+    std::string literalAttrStr;
+    if (literalKind == LiteralBindingKind::Atom) {
```

Comment:
Let's rework this whole block to use `getLiteralRequirements()` (name for `getAllProtocolKinds()` I have suggested) and interleave it using `,` as a separator so the output is going to be `literal: int, collection` in case there are multiple conflict literal requirements which would be very helpful for debugging!